### PR TITLE
Add llama3 chat template

### PR DESCRIPTION
**Pull request by AllentDan**

### CODE REVIEW

File: lmdeploy/model.py [hunk 1]
```diff
@@ -492,6 +492,44 @@ def match(cls, model_path: str) -> Optional[str]:
...
+
+    @classmethod
+    def match(cls, model_path: str) -> Optional[str]:
+        """Return the model_name that was registered to MODELS.
+
+        Args:
+            model_path (str): the model path used for matching.
+        """
+        if 'llama-3' in model_path.lower() or 'llama3' in model_path.lower():
+            if 'instruct' in model_path.lower():
```
Comment (zhyncs) [inline lmdeploy/model.py:529]: Why increase the condition on `instruct`?

Comment (zhyncs) [reply lmdeploy/model.py:529]: Have you tested all four of them?
```
meta-llama/Meta-Llama-3-8B
meta-llama/Meta-Llama-3-8B-Instruct
meta-llama/Meta-Llama-3-70B
meta-llama/Meta-Llama-3-70B-Instruct
```

Comment (AllentDan) [reply lmdeploy/model.py:529]: Thought only instruct model is cappable of chatting. I will remove it.

File: lmdeploy/model.py [hunk 2]
```diff
@@ -492,6 +492,44 @@ def match(cls, model_path: str) -> Optional[str]:
             return 'llama2'
 
 
+@MODELS.register_module(name='llama3')
+class Llama3(BaseChatTemplate):
+    """Chat template of LLaMA3 model."""
+
+    def __init__(self,
+                 system='<|start_header_id|>system<|end_header_id|>\n\n',
```
Comment (zhyncs) [inline lmdeploy/model.py:500]: We should match with https://github.com/meta-llama/llama3/blob/359887376f0aaf30e433f23e25df858d8c2a9833/llama/tokenizer.py#L38-L97

File: lmdeploy/model.py (hunk 1 above)
Comment (AllentDan) [reply lmdeploy/model.py]: > Have you tested all four of them?
> 
> ```
> meta-llama/Meta-Llama-3-8B
> meta-llama/Meta-Llama-3-8B-Instruct
> meta-llama/Meta-Llama-3-70B
> meta-llama/Meta-Llama-3-70B-Instruct
> ```

No, this is a chat template, and I tested `Meta-Llama-3-8B-Instruct`

File: lmdeploy/model.py (hunk 2 above)
Comment (AllentDan) [reply lmdeploy/model.py:500]: Which part did not match?

Comment (AllentDan) [reply lmdeploy/model.py:500]: Seems the tokenizer of llama3 does not add bos token itself. I will update it.

**Comment by zhyncs** [comment]
https://huggingface.co/meta-llama/Llama-2-13b-chat-hf/tree/main

tokenizer.model in the root directory


https://huggingface.co/meta-llama/Meta-Llama-3-8B/tree/main

tokenizer.model in the secondary directory
`original/tokenizer.model`

**Comment by zhyncs** [comment]
> https://huggingface.co/meta-llama/Llama-2-13b-chat-hf/tree/main
> 
> tokenizer.model in the root directory
> 
> https://huggingface.co/meta-llama/Meta-Llama-3-8B/tree/main
> 
> tokenizer.model in the secondary directory `original/tokenizer.model`

It's ok. `tokenizer.json` is in the root directory.

**Comment by zhyncs** [comment]
LGTM

**Review by lvhan028 — APPROVED** [review]

**Pull request merged**